#37 create cerberus infrastructure module

TysonBeck opened the pull request.

TysonBeck committed:
#37 Initial commit of infrastructure for cerberus sftp server

TysonBeck committed:
#37 Variablized LB target group health check interval and set defaults to 30 seconds

TysonBeck committed:
#37 Modified health check interval default and modified healthy threshold count

jeremyodle reviewed (changes requested):
- README needs updated
- Terraform remote data call role names need to be variable driven
- Module `stage` should default to `dev`
- Add `desc_prefix` to Security group ingress/egress rules
- Set ALB/NLB target group health checks to use Terraform variables
- Set ALB/NLB target group health checks to max interval @ min healthy/unhealthy threshold
- Add tags to target groups
- Enable Cross-Zone Load Balancing on NLB
- Add `All ICMP` to EC2 security group for `10.0.0.0/8`
- Limit ALB security group access to EC2 security group to only `443`
- Allow `8443` access to EC2 security group for `10.0.0.0/8`
- Change DNS TTL to `30` seconds
- Ensure all resources have `count` on them (e.g. Does `aws_ssm_association` resource support count?)

TysonBeck committed:
#37 Tightened security groups for EC2 and ALB
Changed DNS TTL to 30 seconds
Ensured all supporting resources had count and consolidated logic to remove unnecessary local variables
Enabled Cross-Zone Load Balancing on NLB
Added tags to supported resources
Add desc_prefix to Security group ingress/egress rules
Added variables for remote state call role names

TysonBeck committed:
#37 Updated readme

jeremyodle commented on README.md:
```diff
@@ -20,6 +20,7 @@ This is a collection of reusable Terraform modules for Gravicore's cloud automat
 | [RDS](aws/rds)                               | A collection of modules for deploying Relational Database Service (RDS) resources.                                                                                                                 |
 | [RDS (PostgreSQL)](aws/rds-postgres)         | A module for deploying [Amazon RDS for PostgreSQL](https://aws.amazon.com/rds/postgresql/) resources.                                                                                              |
 | [ElastiCache (Redis)](aws/elasticache-redis) | A module for deploying an ElastiCache for Redis instance.                                                                                                                                          |
+| [Cerberus](aws/cerberus)                     | A module for deploying the infrastructure resources needed for [Cerberus FTP](https://www.cerberusftp.com/).                                                                                       |
```
Change `[Cerberus]` to `[Cerberus FTP]`

jeremyodle commented on aws/cerberus/cerberus.tf:
```diff
@@ -0,0 +1,582 @@
+    to_port         = "443"
+    protocol        = "6"
+    security_groups = ["${aws_security_group.cerberus_alb.id}"]
+    description     = "${var.desc_prefix} All from ALB"
```
Description is `${var.desc_prefix} HTTPS from ALB`

jeremyodle commented on aws/cerberus/cerberus.tf:
```diff
@@ -0,0 +1,582 @@
+    to_port     = "8443"
+    protocol    = "6"
+    cidr_blocks = ["${var.ingress_sg_cidr}"]
+    description = "${var.desc_prefix} Cerberus web adminitration port from internal"
```
Description is `${var.desc_prefix} Cerberus Web Administration from internal`

jeremyodle commented on aws/cerberus/cerberus.tf:
```diff
@@ -0,0 +1,582 @@
+  }
+
+  ingress {
+    from_port   = 0
```
Allow all ICMP:
```
from_port = -1
to_port = -1
protocol = "icmp"
```

jeremyodle commented on aws/cerberus/cerberus.tf:
```diff
@@ -0,0 +1,582 @@
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+    description = "${var.desc_prefix} EC2 egress"
```
Description is `${var.desc_prefix} Allow All`

jeremyodle commented on aws/cerberus/cerberus.tf:
```diff
@@ -0,0 +1,582 @@
+resource "aws_security_group" "cerberus_ec2" {
+  count       = "${var.create == "true" ? 1 : 0 }"
+  name        = "${local.module_prefix}-ec2"
+  description = "${var.desc_prefix} Allow  traffic from internal and VPN and ALB"
```
Description is `${var.desc_prefix} Allow traffic to Cerberus EC2 application instances`

jeremyodle commented on aws/cerberus/cerberus.tf:
```diff
@@ -0,0 +1,582 @@
+  namespace               = ""
+  stage                   = ""
+  name                    = "${local.stage_prefix}-cerberus"
+  description             = "${join(" ", list(var.desc_prefix, "KMS key for cerberus"))}"
```
Use a capital letter for `Cerberus` in descriptions

jeremyodle commented on aws/cerberus/module.tf:
```diff
@@ -0,0 +1,121 @@
+# ----------------------------------------------------------------------------------------------------------------------
+
+provider "aws" {
+  version = "~> 1.58.0"
```
Try bumping the provider to `2.11.0`

jeremyodle commented on aws/cerberus/module.tf:
```diff
@@ -0,0 +1,121 @@
+
+provider "aws" {
+  alias   = "master"
+  version = "~> 1.58.0"
```
Try bumping the provider to `2.11.0`

jeremyodle commented on aws/cerberus/cerberus.tf:
```diff
@@ -0,0 +1,582 @@
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+    description = "${var.desc_prefix} ALB egress"
```
Description is `${var.desc_prefix} Allow All`

jeremyodle commented on aws/cerberus/cerberus.tf:
```diff
@@ -0,0 +1,582 @@
+}
+
+resource "aws_lb_target_group" "cerberus_alb_target_group" {
+  count = "${var.create && var.enable_https == "true" ? 1 : 0 }"
```
Does this work? I would have assumed this would look like this `${var.create == "true" && var.enable_https == "true" ? 1 : 0 }`

jeremyodle commented on aws/cerberus/cerberus.tf:
```diff
@@ -0,0 +1,582 @@
+}
+
+resource "aws_lb" "cerberus_alb" {
+  count = "${var.create && var.enable_https == "true" ? 1 : 0 }"
```
Does this work? I would have assumed this would look like this ${var.create == "true" && var.enable_https == "true" ? 1 : 0 }

jeremyodle commented on aws/cerberus/cerberus.tf:
```diff
@@ -0,0 +1,582 @@
+  zone_id = "${data.aws_route53_zone.public.zone_id}"
+  name    = "transfer"
+  type    = "CNAME"
+  ttl     = "60"
```
TTL should be `30` seconds

jeremyodle commented on aws/cerberus/cerberus.tf:
```diff
@@ -0,0 +1,582 @@
+    path                = "/login"
+    protocol            = "HTTPS"
+    timeout             = "5"
+    healthy_threshold   = "2"
```
Use variables for `timeout` and `threshold`s.

jeremyodle commented on aws/cerberus/cerberus.tf:
```diff
@@ -0,0 +1,582 @@
+    to_port     = 0
+    protocol    = "ICMP"
+    cidr_blocks = ["10.0.0.0/8"]
+    description = "${var.desc_prefix} ICMP"
```
Description is `${var.desc_prefix} ICMP from internal`

jeremyodle commented on aws/cerberus/cerberus.tf:
```diff
@@ -0,0 +1,582 @@
+  health_check {
+    port                = "traffic-port"
+    protocol            = "TCP"
+    healthy_threshold   = "2"
```
Use variables for `threshold`s

TysonBeck committed:
#37 Updated aws provider version to 2.11
updated README
Updated/Corrected descriptions
Simplified count syntax
Corrected security group icmp port value
Added variables for health check threshold and timout

jeremyodle commented on aws/cerberus/cerberus.tf:
```diff
@@ -0,0 +1,590 @@
+  assume_role_policy = "${data.aws_iam_policy_document.cerberus_ec2.json}"
+}
+
+resource "aws_iam_role_policy_attachment" "cerberus_s3_access" {
```
Remove this policy to not allow access to S3.

jeremyodle commented on aws/cerberus/module.tf:
```diff
@@ -0,0 +1,121 @@
+}
+
+variable "desc_prefix" {
+  default = "Gravicore Module:"
```
Set desc prefix to `Gravicore:`

TysonBeck committed:
#37 Removed S3 access from IAM policy
Changed description prefix from "Gravicore Module" to "Gravicore"

TysonBeck committed:
#37 Corrected sg cidr block from static to variable
removed proxy_protocol_v2 from nlb target group as it is not supported by cerberus ftp
